providers/implementations/signature/slh_dsa_sig.c.in

```diff
@@ -153,7 +219,7 @@ static int slh_dsa_signverify_msg_init(void *vctx, void *vkey,
     PROV_SLH_DSA_CTX *ctx = (PROV_SLH_DSA_CTX *)vctx;
     SLH_DSA_KEY *key = vkey;
 
-    if (!ossl_prov_is_running()
+    if (!slh_dsa_self_check(ossl_slh_dsa_key_get0_libctx(key))
```

Comment:
Why are there multiple calls to the self_check here.. The object is setup by the newctx(). That should be the point at which the self_check runs if it need to.

Reply:
I backed out some of these, specifically I had to back out the checks on keymgmt gen_init because pairwise_fail_test fail test assumes the init always work and the fail is on the keygen ...